---
title: "HVP Innovation Award 2019"
date: "`r format(Sys.time(), '%B %d, %Y')`"
author: 
  - name: Yun (Renee) Zhang
    email: zhangy@jcvi.org
  - name: Aishwarya Mandava
    email: amandava@jcvi.org
  - name: Yu (Max) Qian* (*corresponding)
    email: mqian@jcvi.org
---

<br>

## The Project

Current licensed Hepatitis B vaccines require multiple doses to reach desirable efficacy. Immunity induced by HBV is complicated, which starts from innate immune responses, followed by adaptive responses. The response is a mixture of cell-mediated immunity and humoral immune responses, requiring different types of assays to elucidate. Multi-omics data analytics are esssential for undertanding the cellular mechanisum of the correlate of protection induced by the vaccination. Our objective is to integrate the publicly available studies on Hepatitis B vaccines (HBV) at the NIAID-funded ImmPort database (https://immport.org) and the Human Vaccines Project (HVP) for identification of predictors of HBV immunogenicity at cell population level. Specifically, this HVP innovation project aims to develop and apply innovative computational approaches to integrate and reanalyze multiple types of experiment data from the existing clinical studies on hepatitis B vaccine ENGERIXÂ®-B for identifying cell type-specific immune signatures in response to the vaccination.


### Design of HVP01 study

This is a brief summary of the HVP01 study. [Text to be inserted.] For more details, please see its [clinical trial page](https://clinicaltrials.gov/ct2/show/study/NCT03083158).


### Data availability

In this project, we are going to use data from the following.

- Transcriptome profiling data
    + RNA sequencing of whole blood (bulk RNA-seq)
    + RNA sequencing of single immune cells (scRNA-seq)
- Flow cytometry data
- Hepatitis B surface antibody (anti-HBs) titer value
- Clinical metadata

For data sharing, please contact [which team?]

<br>

## The Report

This report only serves the purpose of depositing code and script for the above project. For all other purposes, please contact us directly.

### Table of contents

```{r echo=FALSE, results='asis'}
# rmd <- Sys.glob("*.Rmd")
# rmd <- rmd[rmd != "index.Rmd"]
rmd <- c("data.Rmd", "bulk.Rmd", "deconv.Rmd")
html <- sub(".Rmd", ".html", rmd)
lines <- lapply(rmd, readLines)
yaml <- lapply(lines, rmarkdown:::parse_yaml_front_matter)
cat("<ul>")
for (i in seq_along(rmd)) {
  cat(paste0("<li><a href='", html[i], "'>", yaml[[i]]$title, "</a><br/>",
             "<code>", rmd[i], "</code>", "</li>"))
}
cat("</ul>")
```

We will be using the following auxiliary packages to generate this set of reports.
```{r package, message=FALSE, warning=FALSE, catch=FALSE}
library(dplyr)
library(kableExtra)
library(ggplot2)
library(gridExtra)
library(grid)
```

<br>

## Acknowledgements

This work is funded by [Text to be inserted.] Richard Scheuermann, Brian Aevermann, UBC, ...


<br>

See our other works at [JCVI](https://www.jcvi.org/).
