---
title: "HVP Innovation Award 2019"
date: "`r format(Sys.time(), '%B %d, %Y')`"
author: 
  - name: Yun (Renee) Zhang
    email: zhangy@jcvi.org
  - name: Aishwarya Mandava
    email: amandava@jcvi.org
  - name: Yu (Max) Qian* (*corresponding)
    email: mqian@jcvi.org
---

<br>

## The Project

Current licensed Hepatitis B vaccines require multiple doses to reach desirable efficacy. Immunity induced by HBV is complicated, which starts from innate immune responses, followed by adaptive responses. The response is a mixture of cell-mediated immunity and humoral immune responses, requiring different types of assays to elucidate. Multi-omics data analytics are esssential for undertanding the cellular mechanisum of the correlate of protection induced by the vaccination. Our objective is to integrate the publicly available studies on Hepatitis B vaccines (HBV) at the NIAID-funded [ImmPort database](https://immport.org) and the Human Vaccines Project (HVP) for identification of predictors of HBV immunogenicity at cell population level. Specifically, this HVP innovation project aims to develop and apply innovative computational approaches to integrate and reanalyze multiple types of experiment data from the existing clinical studies on hepatitis B vaccine ENGERIX-B for identifying cell type-specific immune signatures in response to the vaccination.

### Materials and Data

Datasets used in this project are from two ENGERIX-B vaccine studies. One is publicly available at the NIAID Immunology Database and Analysis Portal: [SDY89](https://www.immport.org). The other is the study [HVP01](https://clinicaltrials.gov/ct2/show/NCT03083158). Both studies have assays on whole blood or peripheral blood mononuclear cell (PBMC) samples from adults (young and old), including flow cytometry for immunophenotyping, microarray or RNA-seq for bulk transcriptomics, and virus neutralization assay for serum antibody titers (anti-HBs). Correlates of protection (anti-HBs) are available in both studies for defining cohorts of responders (according to [Wooden 2018](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6183335), correlate for Engerix-B efficacy is titer >10 mIU/ml). Study HVP01 also has RNA-seq data of single immune cells (scRNA-seq). Both studies have planned 11 visits, to collect different types of assay data before and after three doses of the vaccination. Study HVP01 has both flow cytometry and RNA-seq data for all 15 subjects (N=15) for the first 5 of the 11 visits, before and after the 1st dose of the vaccination. For SDY89, we have both flow cytometry and microarray data for 12 of total 50 subjects (N=12) across 8 of the 11 visits, before and after the first two doses of the vaccination.

This page is for summary of the HVP innovation project, so we focus on the analysis of HVP01 study to demonstrate the analysis procedure and results. Feel free to contact us for the analysis results of data from ImmPort SDY89.

### Design of HVP01 study

Design of the HVP01 study can be found at [clinical trial page](https://clinicaltrials.gov/ct2/show/study/NCT03083158).


### Data availability of HVP01 sudy

In this project, we are going to use data from the following.

- Transcriptome profiling data
    + RNA sequencing of whole blood (bulk RNA-seq)
    + RNA sequencing of single immune cells (scRNA-seq)
- Flow cytometry data
- Hepatitis B surface antibody (anti-HBs) titer value
- Clinical metadata


<br>

## The Report

This report only serves the purpose of depositing code and script for the above project. For all other purposes, please contact us directly.

### Table of contents

```{r echo=FALSE, results='asis'}
# rmd <- Sys.glob("*.Rmd")
# rmd <- rmd[rmd != "index.Rmd"]
rmd <- c("data.Rmd", "bulk.Rmd", "deconv.Rmd")
html <- sub(".Rmd", ".html", rmd)
lines <- lapply(rmd, readLines)
yaml <- lapply(lines, rmarkdown:::parse_yaml_front_matter)
cat("<ul>")
for (i in seq_along(rmd)) {
  cat(paste0("<li><a href='", html[i], "'>", yaml[[i]]$title, "</a><br/>",
             "<code>", rmd[i], "</code>", "</li>"))
}
cat("</ul>")
```

We will be using the following auxiliary packages to generate this set of reports.
```{r package, message=FALSE, warning=FALSE, catch=FALSE}
library(dplyr)
library(kableExtra)
library(ggplot2)
library(gridExtra)
library(grid)
```

<br>

## Acknowledgements

This work is funded by the Human Vaccines Project (HVP). Method development for reanalysis of flow cytometry data from ImmPort studies is partially supported by an NIH/NIAID project (UH2AI132342, PI: Qian). Data and metadata of the HVP01 vaccination study used in the analytics were provided by HVP and Kollmann group of University of British Columbia. Brian Aevermann of Scheuermann group at JCVI provided the single cell RNA-seq data analysis results from the HVP01 study for validating the cell type specific marker genes identified in our data analysis. Collaborative development of the FastMix deconvolutional method used in our transcriptomics analysis was led by Qiu group (Dr. Hao Sun and Dr. Xing Qiu) of University of Rochester. Dr. Richard Scheuermann of JCVI provided guidance in assessement and interpretation of the data analysis results. Mehmet Kuscuoglu of JCVI helped with data/metadata management using LabKey.


<br>

See our other works at [JCVI](https://www.jcvi.org/).
