<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2020-01-21" />

<title>HVP Innovation Award 2019</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">HVP-Innovation-Award-2019</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="data.html">Data Prep</a>
</li>
<li>
  <a href="bulk.html">Bulk RNA-seq Analysis</a>
</li>
<li>
  <a href="deconv.html">Deconvolution Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">HVP Innovation Award 2019</h1>
<h4 class="author">Yun (Renee) Zhang</h4>
<a class="author_email" href="mailto:#"><a href="mailto:zhangy@jcvi.org">zhangy@jcvi.org</a></a>
</address>
<h4 class="author">Aishwarya Mandava</h4>
<a class="author_email" href="mailto:#"><a href="mailto:amandava@jcvi.org">amandava@jcvi.org</a></a>
</address>
<h4 class="author">Yu (Max) Qian* (*corresponding)</h4>
<a class="author_email" href="mailto:#"><a href="mailto:mqian@jcvi.org">mqian@jcvi.org</a></a>
</address>
<h4 class="date">January 21, 2020</h4>

</div>


<p><br></p>
<div id="the-project" class="section level2">
<h2>The Project</h2>
<p>Current licensed Hepatitis B vaccines require multiple doses to reach desirable efficacy. Immunity induced by HBV is complicated, which starts from innate immune responses, followed by adaptive responses. The response is a mixture of cell-mediated immunity and humoral immune responses, requiring different types of assays to elucidate. Multi-omics data analytics are esssential for undertanding the cellular mechanisum of the correlate of protection induced by the vaccination. Our objective is to integrate the publicly available studies on Hepatitis B vaccines (HBV) at the NIAID-funded <a href="https://immport.org">ImmPort database</a> and the Human Vaccines Project (HVP) for identification of predictors of HBV immunogenicity at cell population level. Specifically, this HVP innovation project aims to develop and apply innovative computational approaches to integrate and reanalyze multiple types of experiment data from the existing clinical studies on hepatitis B vaccine ENGERIX-B for identifying cell type-specific immune signatures in response to the vaccination.</p>
<div id="materials-and-data" class="section level3">
<h3>Materials and Data</h3>
<p>Datasets used in this project are from two ENGERIX-B vaccine studies. One is publicly available at the NIAID Immunology Database and Analysis Portal: <a href="https://www.immport.org">SDY89</a>. The other is the study <a href="https://clinicaltrials.gov/ct2/show/NCT03083158">HVP01</a>. Both studies have assays on whole blood or peripheral blood mononuclear cell (PBMC) samples from adults (young and old), including flow cytometry for immunophenotyping, microarray or RNA-seq for bulk transcriptomics, and virus neutralization assay for serum antibody titers (anti-HBs). Correlates of protection (anti-HBs) are available in both studies for defining cohorts of responders (according to <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6183335">Wooden 2018</a>, correlate for Engerix-B efficacy is titer &gt;10 mIU/ml). Study HVP01 also has RNA-seq data of single immune cells (scRNA-seq). Both studies have planned 11 visits, to collect different types of assay data before and after three doses of the vaccination. Study HVP01 has both flow cytometry and RNA-seq data for all 15 subjects (N=15) for the first 5 of the 11 visits, before and after the 1st dose of the vaccination. For SDY89, we have both flow cytometry and microarray data for 12 of total 50 subjects (N=12) across 8 of the 11 visits, before and after the first two doses of the vaccination.</p>
<p>This page is for summary of the HVP innovation project, so we focus on the analysis of HVP01 study to demonstrate the analysis procedure and results. Feel free to contact us for the analysis results of data from ImmPort SDY89.</p>
</div>
<div id="design-of-hvp01-study" class="section level3">
<h3>Design of HVP01 study</h3>
<p>Design of the HVP01 study can be found at <a href="https://clinicaltrials.gov/ct2/show/study/NCT03083158">clinical trial page</a>.</p>
</div>
<div id="data-availability-of-hvp01-sudy" class="section level3">
<h3>Data availability of HVP01 sudy</h3>
<p>In this project, we are going to use data from the following.</p>
<ul>
<li>Transcriptome profiling data
<ul>
<li>RNA sequencing of whole blood (bulk RNA-seq)</li>
<li>RNA sequencing of single immune cells (scRNA-seq)</li>
</ul></li>
<li>Flow cytometry data</li>
<li>Hepatitis B surface antibody (anti-HBs) titer value</li>
<li>Clinical metadata</li>
</ul>
<p><br></p>
</div>
</div>
<div id="the-report" class="section level2">
<h2>The Report</h2>
<p>This report only serves the purpose of depositing code and script for the above project. For all other purposes, please contact us directly.</p>
<div id="table-of-contents" class="section level3">
<h3>Table of contents</h3>
<ul>
<li>
<a href='data.html'>Data Prep</a><br/><code>data.Rmd</code>
</li>
<li>
<a href='bulk.html'>Bulk RNA-seq Analysis</a><br/><code>bulk.Rmd</code>
</li>
<li>
<a href='deconv.html'>Deconvolution Anlaysis</a><br/><code>deconv.Rmd</code>
</li>
</ul>
<p>We will be using the following auxiliary packages to generate this set of reports.</p>
<pre class="r"><code>library(dplyr)
library(kableExtra)
library(ggplot2)
library(gridExtra)
library(grid)</code></pre>
<p><br></p>
</div>
</div>
<div id="acknowledgements" class="section level2">
<h2>Acknowledgements</h2>
<p>The work is funded by the Human Vaccines Project (HVP). Reanalysis of data from ImmPort studies is partially supported by NIAID UH2AI132342. Data and metadata of the HVP01 vaccination study used in the analytics were provided by HVP and Kollmann group of University of British Columbia. Brian Aevermann of Scheuermann group at JCVI provided the single cell RNA-seq data analysis results from the HVP01 study for validating the cell type specific marker genes identified in our data analysis.</p>
<p><br></p>
<p>See our other works at <a href="https://www.jcvi.org/">JCVI</a>.</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
